### PR TITLE
Peers RFC

### PR DESCRIPTION
**Pull request by breerly**
I'd like to make sure that we start from day 1 with a multiple peers strategy. This will enable us to build high-level features like load balancing, rate limiting, and circuit breaking. 

Please review @yarpc/yarpc

### CODE REVIEW

File: peers.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+```yaml
+yarpc:
+  name: myservice
+  outbounds:
+
+    # the following 2 outbounds are explicit
+    - service: users
+      http:
+      peers:
+        - scheme: http
```
Comment (kriskowal) [inline peers.md]: Let’s call the scheme “transport” to eliminate ambiguity.

File: peers.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        ])),
+    },
+)
+rpc.Start()
+```
+
+This approach enables peer-based features like:
+
+* load balancing (TBD)
+* retries (TBD)
```
Comment (kriskowal) [inline peers.md]: retry / deflection
retry speculation

File: peers.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        peers:
+          - host: 127.0.0.1
+            port: 5433
+          - host: 127.0.0.1
+            port: 5455
+
+    # this outbound uses a stock config, which has the multiple peers inside
+    - service: blackbear
+      http:
+        with: sidecar
```
Comment (kriskowal) [inline peers.md]: How are stock configurations threaded?

File: peers.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+```
+rpc := yarpc.New(yarpc.Config{
+    Name: "myservice",
+    Outbounds: transport.Outbounds{
+        ...
+    },
+    LoadBalancingPolicy: balance.NewRoundRobin(),
+    RetryPolicy: retry.NewPolicy(retry.MaxAttempts(7)),
+    CircuitBreaker: circuit.New(),
```
Comment (abhinav) [inline peers.md]: Would we also want per-outbound/per-service policies?

Comment (breerly) [inline peers.md]: Perhaps yeah. I think starting with a global policy probably make more sense?

File: peers.md (hunk 3 above)
Comment (breerly) [inline peers.md]: I'm not sure I get the question. The shape of stock configs is largely TBD.

File: peers.md (hunk 1 above)
Comment (breerly) [inline peers.md]: This is actually scheme here, like `http` vs `https`.

**Comment by abhinav** [comment]
👍

**Pull request merged**